packages/m2m-gateway/test/integration/eServiceTemplates/updatePublishedEServiceTemplateVersionQuotas.test.ts

```diff
@@ -0,0 +1,179 @@
+import { toM2MGatewayEServiceTemplateVersion } from "../../../src/api/eserviceTemplateApiConverter.js";
+
+describe("updatePublishedEServiceTemplateVersionQuotas", () => {
+  const mockVersion: eserviceTemplateApi.EServiceTemplateVersion = {
+    ...getMockedApiEserviceTemplateVersion(),
+    state: "PUBLISHED",
+    voucherLifespan: 0,
+    dailyCallsPerConsumer: 10,
+    dailyCallsTotal: 10,
+  };
+
+  const mockEServiceTemplate: eserviceTemplateApi.EServiceTemplate =
+    getMockedApiEServiceTemplate({
+      versions: [mockVersion, getMockedApiEserviceTemplateVersion()],
+    });
+
+  const mockEServiceTemplateProcessResponse =
+    getMockWithMetadata(mockEServiceTemplate);
+
+  const mockQuotasSeed: m2mGatewayApi.EServiceTemplateVersionQuotasUpdateSeed =
+    {
+      voucherLifespan: 3600,
+      dailyCallsPerConsumer: 1000,
+      dailyCallsTotal: 10000,
+    };
+
+  const mockPatchUpdateQuotas = vi
+    .fn()
+    .mockResolvedValue(mockEServiceTemplateProcessResponse);
+
+  const mockGetEServiceTemplate = vi.fn(
+    mockPollingResponse(mockEServiceTemplateProcessResponse, 1)
+  );
+
+  mockInteropBeClients.eserviceTemplateProcessClient = {
+    getEServiceTemplateById: mockGetEServiceTemplate,
+    updateTemplateVersionQuotas: mockPatchUpdateQuotas,
+  } as unknown as PagoPAInteropBeClients["eserviceTemplateProcessClient"];
+
+  beforeEach(() => {
+    mockPatchUpdateQuotas.mockClear();
+    mockGetEServiceTemplate.mockClear();
+  });
+
+  it("Should succeed and perform service calls", async () => {
+    const result =
+      await eserviceTemplateService.updatePublishedEServiceTemplateVersionQuotas(
+        unsafeBrandId(mockEServiceTemplate.id),
+        unsafeBrandId(mockVersion.id),
+        mockQuotasSeed,
+        getMockM2MAdminAppContext()
+      );
+
+    const expectedM2MEServiceTemplateVersion: m2mGatewayApi.EServiceTemplateVersion =
+      toM2MGatewayEServiceTemplateVersion(mockVersion);
+
+    expect(result).toEqual(expectedM2MEServiceTemplateVersion);
+    expectApiClientPostToHaveBeenCalledWith({
+      mockPost:
+        mockInteropBeClients.eserviceTemplateProcessClient
+          .updateTemplateVersionQuotas,
+      params: {
+        templateId: mockEServiceTemplate.id,
+        templateVersionId: mockVersion.id,
+      },
+      body: mockQuotasSeed,
+    });
+    expectApiClientGetToHaveBeenCalledWith({
+      mockGet:
+        mockInteropBeClients.eserviceTemplateProcessClient
+          .getEServiceTemplateById,
+      params: { templateId: mockEServiceTemplate.id },
+    });
+    expectApiClientGetToHaveBeenNthCalledWith({
+      nthCall: 2,
+      mockGet:
+        mockInteropBeClients.eserviceTemplateProcessClient
+          .getEServiceTemplateById,
+      params: { templateId: mockEServiceTemplate.id },
+    });
+    expect(
+      mockInteropBeClients.eserviceTemplateProcessClient.getEServiceTemplateById
+    ).toHaveBeenCalledTimes(2);
+  });
+
```

Comment:
Could you also add a test for the partial patch logic? Since it's implemented on the M2M side in this case.
I did something similar here: https://github.com/pagopa/interop-be-monorepo/blob/release/2.7.0/packages/m2m-gateway/test/integration/eservices/updatePublishedEServiceDelegation.test.ts#L105-L106
Also, I realized that both API and Integration tests are missing for the corresponding e-service API implemented here #2321 😅 I think I merged without adding tests, I'm gonna do a dedicated PR for that